smartclient-core/src/main/java/org/srg/smartclient/utils/AnnotationUtils.java

```diff
@@ -105,14 +106,88 @@ private static String replaceLast(String text, String regex, String replacement)
             }
         }
 
-        final A ai = effectiveAnnotations.isEmpty() ? null : mergeJoinColumns(annotationClass, effectiveAnnotations);
+        final List<A> ea = effectiveAnnotations.stream().map( a -> {
+            final Map<String, Object> values = new HashMap<>(defaultValues);
+
+            a.values().forEach( av -> {
+                switch (av.kind()) {
+                    case ENUM:
+                        final DotName enumClassName = av.asEnumType();
+                        try {
+                            final Class enumClass = Class.forName(enumClassName.toString());
+                            final Object o = Enum.valueOf(enumClass, av.asEnum());
+                            values.put(av.name(), o);
+                        } catch (ClassNotFoundException e) {
+                            throw new RuntimeException(e);
+                        }
+                        break;
+
+                    default:
+                        values.put(av.name(), av.value());
+                }
+            });
+
+            return RuntimeAnnotations.annotationForMap(annotationClass, values);
+        }).collect(Collectors.toList());
+
+        final A ai = effectiveAnnotations.isEmpty() ? null : mergeAnnotations(annotationClass, ea);
         return ai;
     }
 
+    /**
+     * Compares two maps and returns changes from the right map (that does not exist in the left one).
+     */
+    protected static Map<String, Object> detectChanges(Map<String,Object> left, Map<String,Object> right) {
+        final Map<String, Object> changes = new HashMap<>();
+        for (var e : right.entrySet()) {
+            final String k = e.getKey();
+            final Object rv = e.getValue();
+            Object ev = left.get(k);
+
+            if (rv instanceof Map rm) {
+                if (ev instanceof Annotation a) {
+                    ev = annotationValues(a);
+                }
+                final Map<String, Object> vv = detectChanges((Map)ev, rm);
+
+                if (!vv.isEmpty()) {
+                    changes.put(k, vv);
+                }
+            } else if (ev != rv && !ev.equals(rv)) {
+                changes.put(k, rv);
+            }
+        }
+        return changes;
+    }
+
+    /**
+     * Retrieves all annotation values.
+     */
+    public  static <A extends Annotation> Map<String, Object> annotationValues(A annotation) {
+        final Map<String, Object> values = new HashMap<>();
+
+        final  Method methods[] =  annotation.annotationType().getDeclaredMethods();
+        for (Method m : methods) {
+            Object v = null;
```

Comment:
Seems like you don't use object `v` outside of try block. I think it's better to create and init it in line 173.